lib/trento/clusters.ex

```diff
@@ -181,29 +178,53 @@ defmodule Trento.Clusters do
     end
   end
 
-  defp maybe_request_checks_execution(%{selected_checks: []}), do: :ok
+  defp maybe_request_checks_execution(%ClusterReadModel{selected_checks: []}), do: :ok
 
-  defp maybe_request_checks_execution(%{
+  defp maybe_request_checks_execution(%ClusterReadModel{
          id: cluster_id,
          provider: provider,
          type: cluster_type,
-         details: details,
+         additional_sids: cluster_sids,
          selected_checks: selected_checks
        }) do
     hosts_data =
-      Repo.all(
-        from h in HostReadModel,
-          select: %{host_id: h.id},
-          where: h.cluster_id == ^cluster_id and is_nil(h.deregistered_at)
-      )
+      case cluster_type do
+        :ascs_ers ->
+          Repo.all(
+            from h in HostReadModel,
+              select: %{host_id: h.id},
+              join: a in ApplicationInstanceReadModel,
+              on: h.id == a.host_id,
+              where:
+                h.cluster_id == ^cluster_id and is_nil(h.deregistered_at) and
+                  a.sid in ^cluster_sids
+          )
+
+        _ ->
+          Repo.all(
+            from h in HostReadModel,
+              select: %{host_id: h.id},
+              where: h.cluster_id == ^cluster_id and is_nil(h.deregistered_at)
+          )
+      end
 
     env =
       case cluster_type do
         :ascs_ers ->
+          host_ids = Enum.map(hosts_data, fn h -> h.host_id end)
+
+          sap_system_ids =
```

Comment:
Do you need this query?
You could get the sap system id in the query above including in in the select